components/molecules/SearchMultiSelect.jsx

```diff
@@ -1,289 +1,340 @@
-                            optionIcon="bt-times"
-                            optionLabel="Clear all tags"
-                            size={tagSize}
-                            text={`${numberSelected} Selected`}
-                        />
-                    )}
-                    <input
-                        ref={inputRef}
-                        onFocus={this._openSelect}
-                        onClick={this._openSelect}
-                        onChange={this._updateSearchTerm}
-                        type="text"
-                        placeholder={placeholder}
-                        className={inputClasses}
-                    />
-                    <button
-                        className="gds-search-select__toggle-button -cursor--pointer"
-                        type="button"
-                        onClick={this._toggleSelect}
+    const _removeOption = (index, subIndex) => {
+        const oldOption = options[index];
+        const newOptions = options.slice();
+        newOptions[index] = { ...oldOption, isSelected: false };
+        if (newOptions[index].options) {
+            newOptions[index].options[subIndex].isSelected = false;
+        }
+        update(newOptions);
+    };
+
+    return (
+        <div ref={containerRef} className={rootClass}>
+            <div className="gds-search-select__control">
+                {showTags && (
+                    <Tag
+                        className={tagIndicatorClasses}
+                        context={context}
+                        hasOption
+                        onClick={_toggleTags}
+                        onOptionClick={_clearAll}
+                        optionIcon="bt-times"
+                        optionLabel="Clear all tags"
+                        size={tagSize}
+                        text={`${numberSelected && numberSelected.length} Selected`}
                     />
-                </div>
-                <div className={tagHolderClasses}>
-                    <div className="gds-search-select__tag-overflow">
-                        {options.map(({ name, key, isSelected }, index) => {
-                            if (!isSelected) return;
+                )}
+                <input
+                    ref={inputRef}
+                    onFocus={_openSelect}
+                    onClick={_openSelect}
+                    onChange={_updateSearchTerm}
+                    type="text"
+                    placeholder={placeholder}
+                    className={inputClasses}
+                />
+                <button
+                    className="gds-search-select__toggle-button -cursor--pointer"
+                    type="button"
+                    onClick={_toggleSelect}
+                />
+            </div>
+            <div className={tagHolderClasses}>
+                <div className="gds-search-select__tag-overflow">
+                    {options.map(({ name, key, isSelected, options }, index) => {
```

Comment:
it looks like it's still working but just wanted to make sure.. this PR is changing the structure of the options you pass in to have "value" instead of "key".. will this still work if someone had already been passing in "key"? and it works still if you pass in "value" since key is referenced here?

can you try these steps https://github.com/gumgum/gumdrops/blob/master/CONTRIBUTING.md#local-testing to test it on AM in a place where it's using the options using key, and one using value?

also it's a little hard to test on gumdrops since you have to edit the options manually so would be good to test on a real app!

Reply:
I test it and it works, also did some changes to improve performance